src/content/reference/react-dom/components/form.md

```diff
@@ -0,0 +1,451 @@
+
+#### Caveats {/*caveats*/}
+
+* When a function is passed to `action` or `formAction` the HTTP method will be POST regardless of value of the `method` prop.
+
+---
+
+## Usage {/*usage*/}
+
+### Handling form submission without a server {/*handle-form-submission-on-the-client*/}
+
+Render a `<form>` with a input and submit button. Pass a function to the `action` prop of form to run the function when the form is submitted.
+
+<Sandpack>
+
+```js App.js
+export default function Search() {
+  function search(formData) {
+    const query = formData.get("query");
+    alert(`You searched for '${query}'`);
+  }
+  return (
+    <form action={search}>
+      <input name="query" />
+      <button type="submit">Search</button>
+    </form>
+  );
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-6db7f4209-20231021",
+    "react-dom": "18.3.0-canary-6db7f4209-20231021",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+### Submitting forms without JavaScript {/*submitting-forms-without-javascript*/}
+
+Render a `<form>` with a input and submit button. Pass a function marked with `'use server'` to the `action` prop of form to run the function when the form is submitted. When used with [Server Components](/reference/react/use-client), the form is [progressively enhanced](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement).
+
+```jsx
+import { updateCart } from './lib.js';
+
+function AddToCart({productId}) {
+  async function addToCart(formData) {
+    'use server'
+    const productId = formData.get('productId')
+    await updateCart(productId)
+  }
+  return (
+    <form action={addToCart}>
+        <input type="hidden" name="productId" value={productId} />
+        <button type="submit">Add to Cart</button>
+    </form>
+
+  );
+}
+```
+
+In leiu, of using hidden form fields to provide data to the function passed to the `action` prop of `<form>` you can call the `bind` method on the function passed to the `action` prop of `<form>`.
+
+```jsx
+import { updateCart } from './lib.js';
+
+function AddToCart({productId}) {
+  async function addToCart(productId, formData) {
+    "use server";
+    await updateCart(productId)
+  }
+  const addProductToCart = addToCart.bind(null, productId);
+  return (
+    <form action={addProductToCart}>
+      <button type="submit">Add to Cart</button>
+    </form>
+  );
+}
+```
+
+### Display a pending state during form submission {/*display-a-pending-state-during-form-submission*/}
+To display a pending state while a form is submitting, you can call the `useFormStatus` Hook in a component rendered in a `<form>` and read the `pending` property returned.
```

Comment:
Not sure if it's consistent but it seems wrong to me to capitalize "Hook" here.

Reply:
I think you want the passive "is being submitted" throughout. Forms don't submit, they are submitted.

Reply:
[Capital "H" Hook seems to be the convention on react.dev](https://github.com/search?q=repo%3Areactjs%2Freact.dev+%22%60+hook%22&type=code). I agree it looks a little strange

Fixed the tense issue.

Reply:
We do capitalize Hook and have done so since 2018 but 5 years later it may be time to lowercase, especially since the word "hook" is now uncommon in any other context (whereas people would talk about lifecycle hooks, etc. before).